tests/Microsoft.Data.Analysis.Tests/DataFrame.IOTests.cs

```diff
@@ -104,5 +104,140 @@ Stream GetStream(string streamData)
+            Assert.Equal(7, df.Columns.Count);
+            Assert.Equal("CMT", df["vendor_id"][3]);
+
+            DataFrame reducedRows = DataFrame.LoadCsv(GetStream(data), separator: ';', numberOfRowsToRead: 3);
+            Assert.Equal(3, reducedRows.Rows.Count);
+            Assert.Equal(7, reducedRows.Columns.Count);
+            Assert.Equal("CMT", reducedRows["vendor_id"][2]);
+        }
+
+        [Fact]
+        public void TestReadCsvWithExtraColumnInHeader()
+        {
+            string data = @"vendor_id,rate_code,passenger_count,trip_time_in_secs,trip_distance,payment_type,fare_amount,extra
+CMT,1,1,1271,3.8,CRD,17.5
+CMT,1,1,474,1.5,CRD,8
+CMT,1,1,637,1.4,CRD,8.5
+CMT,1,1,181,0.6,CSH,4.5";
+
+            Stream GetStream(string streamData)
+            {
+                return new MemoryStream(Encoding.Default.GetBytes(streamData));
+            }
+            DataFrame df = DataFrame.LoadCsv(GetStream(data));
+
+            Assert.Equal(4, df.Rows.Count);
+            Assert.Equal(7, df.Columns.Count);
+            Assert.Equal("CMT", df["vendor_id"][3]);
+
+            DataFrame reducedRows = DataFrame.LoadCsv(GetStream(data), numberOfRowsToRead: 3);
+            Assert.Equal(3, reducedRows.Rows.Count);
+            Assert.Equal(7, reducedRows.Columns.Count);
+            Assert.Equal("CMT", reducedRows["vendor_id"][2]);
+        }
+
+        [Fact]
+        public void TestReadCsvWithExtraColumnInRow()
+        {
+            string data = @"vendor_id,rate_code,passenger_count,trip_time_in_secs,trip_distance,payment_type,fare_amount
+CMT,1,1,1271,3.8,CRD,17.5,0
+CMT,1,1,474,1.5,CRD,8,0
+CMT,1,1,637,1.4,CRD,8.5,0
+CMT,1,1,181,0.6,CSH,4.5,0";
+
+            Stream GetStream(string streamData)
+            {
+                return new MemoryStream(Encoding.Default.GetBytes(streamData));
+            }
+
+            Assert.Throws<IndexOutOfRangeException>(() => DataFrame.LoadCsv(GetStream(data)));
+        }
+
+        [Fact]
+        public void TestReadCsvWithLessColumnsInRow()
+        {
+            string data = @"vendor_id,rate_code,passenger_count,trip_time_in_secs,trip_distance,payment_type,fare_amount
+CMT,1,1,1271,3.8,CRD
+CMT,1,1,474,1.5,CRD
+CMT,1,1,637,1.4,CRD
+CMT,1,1,181,0.6,CSH";
+
+            Stream GetStream(string streamData)
+            {
+                return new MemoryStream(Encoding.Default.GetBytes(streamData));
+            }
+
+            DataFrame df = DataFrame.LoadCsv(GetStream(data));
+            Assert.Equal(4, df.Rows.Count);
+            Assert.Equal(6, df.Columns.Count);
+            Assert.Equal("CMT", df["vendor_id"][3]);
+
+            DataFrame reducedRows = DataFrame.LoadCsv(GetStream(data), numberOfRowsToRead: 3);
+            Assert.Equal(3, reducedRows.Rows.Count);
+            Assert.Equal(6, reducedRows.Columns.Count);
+            Assert.Equal("CMT", reducedRows["vendor_id"][2]);
+
+        }
+
+        [Fact]
+        public void TestReadCsvWithCommaInString()
```

Comment:
Maybe remove this test? We actually have an issue filed against this: https://github.com/dotnet/corefxlab/issues/2787. We want to be able to handles values with `,` inside quotes.

Reply:
Sure. Or would it be better to ignore this one for when the other issue gets fixed it can then be used for it? Just a thought :)

Reply:
Alright sure, we can keep it. I don't really have a strong preference here.

Reply:
Skipped the test and removed the `Assert.Throws`. Not quite sure what the expected output should be in this case so I left the rest of the test as is, so it would still need to be updated.